tripal/src/Entity/TripalEntity.php

```diff
@@ -126,21 +134,21 @@ public function label() {
   /**
    * {@inheritdoc}
    */
-  public function setTitle($title = NULL, $cache = []) {
-
-    // Get the bundle object.
-    if (isset($cache['bundle'])) {
-      $bundle = $cache['bundle'];
-    }
-    else {
+  public function setTitle($title = NULL) {
```

Comment:
I think we should keep the $cache parameter here and the associated support for caching the bundle. This is helpful for performance reasons as there may be workflows where we already have the bundle and do not want to look it up repeatedly. For example, in a workflow where you are publishing the content for a single bundle. You may have used the bundle object in (1) the publishing service, (2) setting title, (3) setting URL alias. Your changes mean it would be loaded 3 times but leaving in the cache and passing it through reduces it to us loading it once.

Reply:
I only removed this because it was not implemented and didn't do anything 🙈
I will work on adding it back...

Reply:
cache is made a class variable - let me know if this is reasonable!